scan-framework/src/main/java/com/getbouncer/scan/framework/util/Layout.kt

```diff
@@ -81,4 +81,92 @@ fun RectF.centerScaled(scaleX: Float, scaleY: Float): RectF {
     )
 }
 
+/**
+ * Scales up a rect
+ */
+fun Rect.scale(scaleX: Float, scaleY: Float) = Rect(
+    this.centerX() - (this.width() * scaleX / 2).toInt(),
+    this.centerY() - (this.height() * scaleY / 2).toInt(),
+    this.centerX() + (this.width() * scaleX / 2).toInt(),
+    this.centerY() + (this.height() * scaleY / 2).toInt()
+)
+
+/**
+ * Creates a rectangle of the given aspect ratio, so that the original rect fits directly inside
+ */
+fun Rect.expandToAspectRatio(widthRatio: Int, heightRatio: Int): Rect {
```

Comment:
There is a similar method named `maxApsectRatioInSize`, which takes a given `Size` and aspect ratio as a `Float` (x/y) and creates a size of aspect ratio that fits within the given size.

Let's rename this similarly `minAspectRatioSurrounding(area: Size, aspectRatio: Float): Size` and then call it like this:

```
minAspectRatioSurrounding(rect.size(), width.toFloat() / height).centeredOn(rect)
```